### ISSUE TITLE
issues running the pulsechain_validator_setup.sh

### ISSUE THREAD

**Issue by machiavegli91** [rhmaxdotorg/pulsechain-validator#41]
I have now invested around 80 hours. learning alot about linux. i have reinstalled a fresh linux already 6-7 times with different setups. the problem persists no matter what.

Current setup looks like this: 
Intel Nuc with an i7 and a 4 tb ssd
Ubuntu 22.04 fresh install

installed the following without modification:
ssh
fail2ban

cloned the github repo on a non-root user with sudo rights. 
executed the pulsechain_validator_setup.sh from that user and i am not able to get it running...

here are the error messages:

link: github.com/fjl/memsize: invalid reference to runtime.stopTheWorld
util.go:48: exit status 1
exit status 1
make: *** [Makefile:13: geth] Error 1

------------------------------------------

error[E0282]: type annotations needed for `Box<_>`
  --> /home/node/.cargo/registry/src/index.crates.io-6f17d22bba15001f/time-0.3.34/src/format_description/parse/mod.rs:83:9
   |
83 |     let items = format_items
   |         ^^^^^
...
86 |     Ok(items.into())
   |              ---- type must be known at this point
   |
   = note: this is an inference error on crate `time` caused by an API change in Rust 1.80.0; update `time` to version `>=0.3.35` by calling `cargo update`

   Compiling tokio v1.36.0
   Compiling impl-codec v0.6.0
   Compiling scale-info v2.10.0
   Compiling asynchronous-codec v0.6.2
   Compiling slog v2.7.0
   Compiling fixed-hash v0.8.0
   Compiling bitflags v2.4.2
   Compiling unsigned-varint v0.7.2
For more information about this error, try `rustc --explain E0282`.
error: could not compile `time` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: failed to compile `lighthouse v2.5.0 (/opt/lighthouse/lighthouse)`, intermediate artifacts can be found at `/opt/lighthouse/target`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
make: *** [Makefile:51: install] Error 101

i already changed the script to try out different versions of rust...i tried everything in my skillset. my brain is fried. i was sitting today 9 hours in front of it. 

can anybody help?

**Comment by Sipheren** [comment]
Not sure if its the same, it sounds like it is related to my issue here: https://github.com/rhmaxdotorg/pulsechain-validator/issues/38

I had to manually update cargo after the install fails, something about it doesnt update correctly and keeps causing new setups or updates to not work.

Hopefully something in my thread will help.

**Comment by machiavegli91** [comment]
Alright,

so I ran the update script after running the setup script and then recompiled lighthouse and after it i was able to import the validator keys. 
Right now it is syncing with the blockchain. I am running the testnet now and going mainnet once online. 

I kind of wasted really a lot of time because of the obsolete setup script. 
At least it should be noted in the guide that this is the way to go now @rhmaxdotorg 

1. Run the pulsechain-validator-script.sh
2. Run the update-client.sh
3. Recompile Lighthouse
4. Import Keys
5. Start beacon and validator clients

I'll keep you updated if everything works  out fine. 
Thanks for the help @Sipheren !

**Comment by rhmaxdotorg** [comment]
Hey @machiavegli91, glad you got it figured out, wish it was a bit easier of course.

I had made updates to the update-client script back when @Sipheren to fix, if I remember correctly, some new rust/go dependencies breaking changes, but I didn't think there would be any issues with new validator setup (can't remember if I tried a new setup back then though).

https://github.com/rhmaxdotorg/pulsechain-validator/issues/38

So I'm not sure if anyone is running into this now, but if there is change that can avoid it going forward, feel free to share a code diff or PR for the setup script and I can take a look. It takes some effort to test these things, so I try to not make big changes to the setup script without some testing as it can break other things.

**Comment by machiavegli91** [comment]
Okay,
 the testnet client fully synced. So i executed the reset script, changed the setup script to mainnet and ran everything again. this time i did not get error messages after the scripts and i could import the validator keys without recompiling lighthouse.
 
 However when checking the status of lighthouse it was running but i got some errors. I tried to turn off and recompile but it seems it broke everything. i will try again and keep you updated...

**Comment by machiavegli91** [comment]
Update:

It looks like i still had the "old" testnet key in the same folder of my vm. I guess, not sure though something went sideways with that. 
Anyways. It looks like I am syncing with Mainnet now. Once Synced ill close the issue.
@rhmaxdotorg We should explain for newbies in the README that after running the setup script they need to run the update client script, otherwise they will be in despair like i was xD

By the way: Has anyone set this up in a docker? I would like to be able to increase the amount of validators in future without doing everything again.

**Comment by rhmaxdotorg** [comment]
> Update:
> 
> It looks like i still had the "old" testnet key in the same folder of my vm. I guess, not sure though something went sideways with that. Anyways. It looks like I am syncing with Mainnet now. Once Synced ill close the issue. @rhmaxdotorg We should explain for newbies in the README that after running the setup script they need to run the update client script, otherwise they will be in despair like i was xD

As I mentioned, happy to take a look at diffs or PRs as it's not easy to test changes for the main setup script. I'm also not sure this is happening for everyone else, so if someone wants to take a look, suggest a fix and test it, that would make it a lot easier to get the changes to the setup script in.

> By the way: Has anyone set this up in a docker? I would like to be able to increase the amount of validators in future without doing everything again.

There are other scripts/guides that use docker, but I explicitly wanted to avoid it as I think there are pros/cons vs running it in a container.

Glad you got it figured out!

If nothing further on this, feel free to close the issue.

**Comment by machiavegli91** [comment]
Hey guys,
I ran the Pulsechain.mypinata.cloud and made everything there.
I just made the deposit. All transactions went through. While waiting for the web interface to accept the last deposit ( although in the block explorer it already went through ) i unfortuntely refreshed the internet site and everything is gone.

I have checked everything double and triple. The transactions went through and the validator is running.
What do I do now? I have no clue how to check the validator.
I went to https://beacon.pulsechain.com/validators....there i get redirected to https://validatordashboard.com/
I made an account there and imported the validators via deposit.json. 

I cannot see them. I guess I need to wait for some time so they come online?

Is there anything else I can do?
